### PR TITLE
[Alerts] Add abstract methods and fix imports and type hints

### PR DESCRIPTION
**Pull request by TomerShor**
Some improvements to the alert related db methods that align alerts with the rest of the resources (and improves QOL for developers).

### CODE REVIEW

**Review by roei3000b — COMMENTED** [review]
Looks good, small question:
Why did you choose only create this function on the base class? what about `store_alert_template` for example?

File: server/api/db/base.py [hunk 1]
```diff
@@ -835,13 +835,58 @@ def list_alert_templates(self, session) -> list[mlrun.common.schemas.AlertTempla
...
+        pass
+
+    @abstractmethod
+    def store_alert_state(
+        self,
+        session,
+        project: str,
+        name: str,
+        last_updated: datetime,
+        count: typing.Union[int, None] = None,
```
Comment (rokatyy) [inline server/api/db/base.py]: Why not `Optional[int] = None`?

**Comment by TomerShor** [comment]
> Looks good, small question: Why did you choose only create this function on the base class? what about `store_alert_template` for example?

@roei3000b Some functions already exist in the base class (like `store_alert_template`), I only added the ones that were missing.

File: server/api/db/base.py (hunk 1 above)
Comment (TomerShor) [reply server/api/db/base.py]: No reason, changed it.

**Review by rokatyy — APPROVED** [review]
Looks good!

**Review by roei3000b — APPROVED** [review]

**Review by Yacouby — APPROVED** [review]

**Review by liranbg — APPROVED** [review]

**Pull request merged**